extensions/amp-nexxtv-player/0.1/amp-nexxtv-player.js

```diff
@@ -0,0 +1,229 @@
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {user} from '../../../src/log';
+import {
+  installVideoManagerForDoc,
+} from '../../../src/service/video-manager-impl';
+import {removeElement} from '../../../src/dom';
+import {isObject} from '../../../src/types';
+import {tryParseJson} from '../../../src/json';
+import {listen} from '../../../src/event-helper';
+import {VideoEvents} from '../../../src/video-interface';
+import {videoManagerForDoc} from '../../../src/video-manager';
+
+/**
+ * @implements {../../../src/video-interface.VideoInterface}
+ */
+class AmpNexxtvPlayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?string} */
+    this.videoIframeSrc_ = null;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url(this.getVideoIframeSrc_(), opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.playerReadyPromise_ = new Promise(resolve => {
+      this.playerReadyResolver_ = resolve;
+    });
+
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    this.applyFillContent(iframe);
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+
+    this.element.appendChild(iframe);
+
+    installVideoManagerForDoc(this.element);
+    videoManagerForDoc(this.element).register(this);
+  }
+
+  getVideoIframeSrc_() {
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+
+    const mediaId = user().assert(
+      this.element.getAttribute('data-mediaid'),
+      'The data-mediaid attribute is required for <amp-nexxtv-player> %s',
+      this.element);
+
+    const client = user().assert(this.element.getAttribute('data-client'),
+      'The data-client attribute is required for <amp-nexxtv-player> %s',
+      this.element);
+
+    const start = this.element.getAttribute('data-seek-to') || 0;
+    const mode = this.element.getAttribute('data-mode') || 'static';
+    const streamtype = this.element.getAttribute('data-streamtype') || 'video';
+    const origin = this.element.getAttribute('data-origin')
+      || 'https://embed.nexx.cloud/';
+
+    let src = '';
+    src += origin;
+
+    if (streamtype !== 'video') {
+      src += `${encodeURIComponent(streamtype)}/`;
+    }
+
+    src += `${encodeURIComponent(client)}/`;
+    src += `${encodeURIComponent(mediaId)}`;
+    src += `?start=${encodeURIComponent(String(start))}`;
```

Comment:
Nit: we can avoid the string cast by using `'0'` https://github.com/ampproject/amphtml/pull/7816/files#diff-785b3e6c303d2f26c285316e592444f9R100.